WordPress/Classes/ViewRelated/Post/PostSettingsViewController+JetpackSocial.swift

```diff
@@ -36,26 +36,31 @@ extension PostSettingsViewController {
         let isJetpackSocialEnabled = FeatureFlag.jetpackSocial.enabled
         let blogSupportsPublicize = apost.blog.supportsPublicize()
         let blogHasConnections = publicizeConnections.count > 0
-        // TODO: Check if there's a share limit
+        let isSocialSharingLimited = apost.blog.isSocialSharingLimited
+        let blogHasPublicizeInfo = apost.blog.publicizeInfo != nil
 
         return isJetpackSocialEnabled
         && blogSupportsPublicize
         && blogHasConnections
+        && isSocialSharingLimited
+        && blogHasPublicizeInfo
```

Comment:
If the above suggestion is accepted, don't forget to change these to:

```suggestion
        && blogHasSharingLimit
```